template-processors/base/bin/processParameters.sh

```diff
@@ -21,45 +21,48 @@ echo "Processing Parameters"
 
 FOLDERS=""
 
-if [ -e ${CLONED_PARAMETER_GIT_DIR}/hiera.lst ] ; then
+if [ -e "${CLONED_PARAMETER_GIT_DIR}/hierarchy.lst" ] ; then
   echo "Generating hierarchy"
   # a hierarchy list was provided, so lets process it
-  while read -r DIR
-    do
+  while read -r DIR ; do
       # remove everything after # to allow comments and nuke all whitespaces
-      DIR="$(echo -e "${DIR%%#*}" | tr -d '[:space:]')"
+      DIR="$(echo "${DIR}" | sed -e 's/[[:space:]]*#.*//')"
 
       if [ -n "${DIR}" ]; then
         # add the folder to the end of the list
-        FOLDERS="${FOLDERS} ${CLONED_PARAMETER_GIT_DIR}/${DIR}"
+        DIR="$(cd ${CLONED_PARAMETER_GIT_DIR}/${DIR} ; pwd)"
+        FOLDERS="${FOLDERS} ${DIR}"
       fi
-    done
-fi <<< $(envsubst < ${CLONED_PARAMETER_GIT_DIR}/hiera.lst)
-
-# Add the current folder as the last one to process
-FOLDERS="${FOLDERS} ${CLONED_PARAMETER_GIT_DIR}"
-
-echo "Going to process parameters file in the following folder(s): ${FOLDERS}"
+    done <<< $(envsubst < ${CLONED_PARAMETER_GIT_DIR}/hierarchy.lst)
```

Comment:
Still the same questions as before:

1. Please mention the cool envsubst feature in the readme! we don't have to be shy towards our users with how fancy we are! :] ideally, it might be cool to have it shown in the examples there
2. Please wrap env vars in `"` — including the `$(...)` expression itself
3. Would it still work if we moved the envsubst before the loop, e.g. like this:

        cat "$CLONED_PARAMETER_GIT_DIR/hierarchy.lst" | envsubst | while IFS= read -r DIR; do
            ...
        done

     to make the data flow more in line with the order of reading the script?

Reply:
1. Already done that previously. Will enhance a bit. https://github.com/KohlsTechnology/eunomia/tree/basic_hiera#upcoming-features
2. Done
3. Unfortunately that doesn't work, because it'll start it in a sub-shell and I won't be able to modify the FOLDERS variable. I had it that way before